### PR TITLE
Feature/add comment

### PR DESCRIPTION
**Pull request by KSummers16**
adding a component with a form to add comments to a post

This feature will allow a customer to write a comment when they are looking at a detailed page of a particular post. 

NewComment.py component that is displayed in the PostDetails.py component. It shows an input field, and anything that the user writes in the input field will be saved, as well as the user's id and the id of the post. 

To test: (make sure that your database is running). Click on a particular post, then there will be a "New Comment" section. Put a comment in the input field and then click submit. Once you are shown a new empty input field then you can check postman to see if your comment was added (the ability to see comments on the page will come with a later ticket)

### CODE REVIEW

**Review by HewsonSchulz — CHANGES_REQUESTED** [review]
The functionality is there, and when I submit a comment, it does post it to the database correctly. However, there are a couple things that should be changed.
- The ticket states that "the creation `datetime` should be automatically set to the current date and time". Although this is not included in the ERD, Steve has told us that the ticket info takes the highest priority, and that the ERD/Wireframe are outdated. This means that each comment should have a `date_created` or something similar. This can be fully implemented on the backend, and has already been done before with the `users` creation (_see `user.py`, `create_user`_).
- The ticket also states that "the user should be redirected to the new Comments list page for the related Post". It's totally fine that this functionality is going to be implemented in a future PR. But if that's the case, then you should unlink the ticket from the development tab, because otherwise, it will automatically mark the ticket as closed when this PR is merged.
- Lastly, the styling is broken for posts that contain a header image (_see attached screenshots_).

Screen Shot 1
Screen Shot 2

**Comment by KSummers16** [comment]
So the header images aren't part of my component. Those were there from the "PostDetails" component. But I will fix the rest for sure

**Comment by HewsonSchulz** [comment]
> So the header images aren't part of my component. Those were there from the "PostDetails" component. But I will fix the rest for sure

I understand that this was not part of your ticket, but your branch currently has broken this styling. This will need to be fixed back how it was before merging. This is what it currently looks like in the `main` branch:

Screen Shot 2024-03-14 at 09 42 46

File: src/components/views/ApplicationViews.js [hunk 1]
```diff
@@ -80,7 +101,10 @@ export const ApplicationViews = () => {
         }
       />
 
-      <Route path="*" element={<Navigate to={"/"} state={{ from: location }} replace />} />
+      {/* <Route
+        path="*"
+        element={<Navigate to={"/"} state={{ from: location }} replace />}
+      /> */}
```
Comment (HewsonSchulz) [inline src/components/views/ApplicationViews.js:107]: This should not be commented out, as it allows users to access pages that do not exist.

File: src/components/views/ApplicationViews.js [hunk 2]
```diff
@@ -34,26 +36,45 @@ export const ApplicationViews = () => {
...
-          <Route path="create" element={<CreateTagForm />} />
-        </Route>
+        }
+      >
+        <Route index element={<>Rare - Home Page</>} />{" "}
+        {/* home page will go here */}
+        <Route path="/category-list" element={<CategoryList />} />
+        <Route path="/createCategory" element={<CategoryForm />} />
+        <Route path="/tags" element={<TagList />} />
+        <Route path="/tags/create" element={<CreateTagForm />} />
```
Comment (HewsonSchulz) [inline src/components/views/ApplicationViews.js:52]: I noticed that I cannot currently access the **Category Manager** view, or the **Category Creation** view. Upon further inspection, it looks like you may have accepted the wrong result when dealing with merge conflicts in `ApplicationViews.js`. The current implementation of `main` includes nested routes for `/categories` and `/tags` in order to support the `/tags/create` and `/categories/create` views. But in your branch, this is what I see when looking at the file changes:

Screen Shot

Lines 42-49 (red) should replace lines 49-52 (green) in order to fix this, and revert this change.

**Review by HewsonSchulz — CHANGES_REQUESTED** [review]
The functionality is all there and seems to work correctly. However, there may have been a mistaken backwards merge when resolving merge conflicts.

File: src/components/views/ApplicationViews.js [hunk 3]
```diff
@@ -34,26 +36,48 @@ export const ApplicationViews = () => {
...
+        <Route index element={<>Rare - Home Page</>} />{" "}
+        {/* home page will go here */}
         <Route path="/categories">
           <Route index element={<CategoryList />} />
           <Route path="create" element={<CategoryForm />} />
         </Route>
+        <Route path="/posts" element={<PostsList />} />
         <Route path="/tags">
           <Route index element={<TagList />} />
-          <Route path="create" element={<CreateTagForm />} />
```
Comment (HewsonSchulz) [inline src/components/views/ApplicationViews.js:48]: This line is still needed.

**Review by HewsonSchulz — CHANGES_REQUESTED** [review]
The functionality is all there and seems to work correctly. However, when attempting to access the **Tag Creation** page, I am redirected to the home page.

**Review by HewsonSchulz — CHANGES_REQUESTED** [review]
Screen Shot

File: src/components/views/posts/PostDetails.js [hunk 4]
```diff
@@ -25,17 +26,34 @@ export const PostDetails = () => {
...
           </div>
           <div className="post-details__content-b">
-            <h3 className="post__date">Published on {formatDate(post.publication_date)}</h3>
+            <h3 className="post-details__date">
+              Published on {formatDate(post.publication_date)}
+            </h3>
+
+            <h3 className="post__date">
+              Published on {formatDate(post.publication_date)}
+            </h3>
```
[the remark this answers is not in the document]
Comment (KSummers16) [reply src/components/views/posts/PostDetails.js:36]: Fixed that now

**Review by HewsonSchulz — APPROVED** [review]
All works as intended.

**Pull request merged**